app/parsers/json.py

```diff
@@ -0,0 +1,41 @@
+from app.objects.c_relationship import Relationship
+from plugins.stockpile.app.parsers.base_parser import BaseParser
+import json
+
+
+class Parser(BaseParser):
+
+    def parse(self, blob):
+        relationships = []
+        try:
+            json_obj = json.loads(blob)
+        except Exception as error:
+            self.log.warning("Output not JSON, use a different parser")
+            return []
+        for mp in self.mappers:
+            json_key = mp.misc.get('json_key')
+            if json_key is None:
+                self.log.warning("JSON Parser not given a 'json_key', not parsing")
+                continue
+            for match in self.get_vals_from_json(json_obj, json_key):
+                source = self.set_value(mp.source, match, self.used_facts)
+                target = self.set_value(mp.target, match, self.used_facts)
+                relationships.append(
+                    Relationship(source=(mp.source, source),
+                                 edge=mp.edge,
+                                 target=(mp.target, target))
+                )
+        return relationships
+
+    def get_vals_from_json(self, json_obj, key):

```

Comment:
rename and make private

Reply:
rename or add docstring describing its behavior, it gets all values for key recursively or it gets first value, etc.